MobiFlight/Joysticks/JoystickManager.cs

```diff
@@ -77,18 +114,8 @@ public void Connect(IntPtr Handle)
                 if (!IsSupportedDeviceType(d)) continue;
 
                 MobiFlight.Joystick js;
-                if (d.InstanceName == "Bravo Throttle Quadrant")
-                {
-                    js = new Joysticks.HoneycombBravo(new SharpDX.DirectInput.Joystick(di, d.InstanceGuid));
-                } else if (d.InstanceName == "Saitek Aviator Stick")
-                {
-                    js = new Joysticks.SaitekAviatorStick(new SharpDX.DirectInput.Joystick(di, d.InstanceGuid));
-                }
-                else
-                {
-                    js = new Joystick(new SharpDX.DirectInput.Joystick(di, d.InstanceGuid));
-                }
-                        
+
+                js = new Joystick(new SharpDX.DirectInput.Joystick(di, d.InstanceGuid), GetDefinitionByInstanceName(d.InstanceName));                        
```

Comment:
All the old Joystick creation code simplifies down to this single line. Pass in the looked up definition and everything just works like it used to!